src/native/corehost/corehost.cpp

```diff
@@ -40,6 +40,19 @@
 #define EMBED_HASH_LO_PART_UTF8 "74e592c2fa383d4a3960714caef0c4f2"
 #define EMBED_HASH_FULL_UTF8    (EMBED_HASH_HI_PART_UTF8 EMBED_HASH_LO_PART_UTF8) // NUL terminated
 
+// This is a workaround for a compiler workaround that
+// causes issues with inserting multiple static strings.
+// See https://github.com/dotnet/runtime/issues/109611 for more details.
```

Comment:
suggestion for this comment:

```
// This avoids compiler optimization which cause EMBED_HASH_HI_PART_UTF8 EMBED_HASH_LO_PART_UTF8 to be placed adjacent causing them to match EMBED_HASH_FULL_UTF8 when searched for replacing.
// See https://github.com/dotnet/runtime/issues/109611 for more details.
```

Reply:
In my analysis, part of the problem is also avx. That causes the compiler to produce multiple copies of the string (and placed in different ELF sections, one of which is for avx operations only). This custom memcmp gets rid of avx usage, which from what I can tell, convinces the compiler to not generate the extra copy.

Reply:
Yes, under `x86-64-v3` the compiler optimizations for memcmp cause copies of these parts that are placed adjacent.